### ISSUE TITLE
[Bug] 3.24: @ember/render-modifiers causes rendering assertion

### ISSUE THREAD

**Issue by NullVoxPopuli** [emberjs/ember.js#19332]
### 🐞 Describe the Bug
When using `{{did-insert}}` or `{{did-update}}`, etc from https://github.com/emberjs/ember-render-modifiers
(which uses the old capabilities API),
the rendering assertion is thrown: 

```
Error: You attempted to update `__tag_value__` on `Tag`, but it had already been used previously in the same computation.  
       Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, 
       and performance issues, and is not supported.
// or
Error: You attempted to update `propertyName` on `<SomeObject:ember1796>`, but it had already been used previously in the same computation. 
  Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, 
  and performance issues, and is not supported.
```

### 🔬 Minimal Reproduction
TODO

### 😕 Actual Behavior
There is a rendering assertion

### 🤔 Expected Behavior
`@ember/render-modifiers`' modifiers should not throw rendering assertions

### 🌍 Environment

- Ember: - 3.24
- OS: - any
- Browser: - any

### ➕ Additional Context
Does not occur in ember 3.20
Discord discussion: https://discord.com/channels/480462759797063690/483601670685720591/797146556264480828

**Comment by rwjblue** [comment]
Is there a reproduction here?

**Comment by NullVoxPopuli** [comment]
Been meaning to make one! Soooooon

**Comment by achambers** [comment]
I'm seeing this when upgrading from 3.14 -> 3.15. Will also try and get a minimal repro together.

**Comment by oliverlj** [comment]
https://gitlab.com/oliverlj/19332

**Comment by bertdeblock** [comment]
@oliverlj I'm not familiar with the original issue reported here, but the error you have is expected behaviour AFAICT. You are reading and mutating the same tracked property in your `did-update` handler which is not allowed: `this.active = !this.active;`.

**Comment by oliverlj** [comment]
@bertdeblock Thanks for your comment. Code is inspired from ember-bootstrap:
https://github.com/kaliber5/ember-bootstrap/blob/master/addon/components/bs-collapse.js

in the _onCollapsedChange method,
active is read and mutaded also.

**Comment by oliverlj** [comment]
Error in my test case, is from ember 3.22, if this is an error

**Comment by bertdeblock** [comment]
> active is read and mutated also.

Yup, which is what the error is trying to prevent. If you read a tracked prop first and then mutate it during a single render, the initial value read won't be correct anymore.

**Comment by johanrd** [comment]
@NullVoxPopuli this can perhaps be retired? seems relevant only for the legacy '3.13' modifier capabilities, which left modifier execution outside the tracking frame until 3.24 plugged the hole? 

- Only '3.22' is accepted; anything else asserts at registration: https://github.com/emberjs/ember.js/blob/53dc03928c8f6bd1eb2c467e904bb498c11a9cb3/packages/%40glimmer/manager/lib/public/modifier.ts#L30-L34
- The '3.13' manager was removed on main on 2021-08-16 (f4c8c69495, "Remove deprecated Modifier Manager version 3.13").

**Comment by NullVoxPopuli** [comment]
Yup. Tyty
